### ISSUE TITLE
share library support

### ISSUE THREAD

**Issue by blshkv** [REDasmOrg/REDasm#28]
Hi, I'm trying to package this tool for Pentoo Linux and have suggestions.

You have released REDasm however, it is not really useful. The current build does not shared libraries and you seem to expect a user to download a random version of QHexView, REDasm-Library and Castone into correct directories and compile everything staticaly in one directory.

The better way would be to use shared libraries. You would need to release dependanceis (QHexView and lib), search for them (including capstone) in a system (not a local folder) and compile as shared libraries.

**Comment by Dax89** [comment]
Mmmmh, QHexView, REDasm-Library and Capstone are submodules that points to a specific version.

* REDasm-Library contains a statically compiled version of capstone.
* REDasm contains the source code of QHexView.

Every REDasm release is unit-tested with a specific version of those libraries.

See here for info: https://github.com/REDasmOrg/REDasm/blob/master/COMPILE.md

**Comment by blshkv** [comment]
well, anyway, my point is that I already have capstone in my system. But each and every single tool is trying to install its own copy of it. Can you change the logic to use a system capstone library first if it is already available?

**Comment by Dax89** [comment]
> well, anyway, my point is that I already have capstone in my system. But many tools are trying to install its own (static) copy of capstone. Can you change the logic to use a system capstone library first if the required version is already available?

Yes, I can, but they can create issues during the analysis, however it's not a problem.

> The next thing, is to follow FHS, https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard
The LibREDasm.so should be installed into /usr/lib{64}/, the binary into /usr/bin
Could you fix CMakeLists.txt to install it properly? The current "install" section is broken: it doesn't install either of these two, but installs capstone headers for no reason ;-)

Ok, no problem! :+1:
